### PR TITLE
Azure OpenAI models: make `endpoint` parameter required

### PR DESCRIPTION
**Pull request by sgrebnov**
# 🗣 Description

Azure OpenAI models: make `endpoint` parameter required

Note: documentation link will be available after upcoming docs release: https://github.com/spiceai/docs/pull/683

### CODE REVIEW

File: crates/runtime/src/model/chat.rs [hunk 1]
```diff
@@ -191,7 +191,7 @@ fn azure(
 ) -> Result<Box<dyn Chat>, LlmError> {
     let Some(model_name) = model_id else {
         return Err(LlmError::FailedToLoadModel {
-            source: format!("For Azure model '{model_name}', model id must be specified in `from:azure:<model_id>`.").into(),
+            source: format!("For Azure model '{model_name}', model id must be specified in `from:azure:<model_id>`.\nFor details on model configuration, visit https://docs.spiceai.org/components/models/azure").into(),
```
Comment (lukekim) [inline crates/runtime/src/model/chat.rs]:
```suggestion
            source: format!(
    "Azure model '{model_name}' requires a model ID in the format `from:azure:<model_id>`. See https://docs.spiceai.org/components/models/azure for details."
).into(),
```

File: crates/runtime/src/model/chat.rs [hunk 2]
```diff
@@ -200,10 +200,19 @@ fn azure(
     let api_key = extract_secret!(params, "azure_api_key");
     let entra_token = extract_secret!(params, "azure_entra_token");
 
+    if api_base.is_none() {
+        return Err(LlmError::FailedToLoadModel {
+            source: format!(
+                "Azure model '{model_name}' requires 'endpoint' parameter.\nFor details on model configuration, visit https://docs.spiceai.org/components/models/azure"
+            )
+            .into(),
```
Comment (lukekim) [inline crates/runtime/src/model/chat.rs]:
```suggestion
            source: format!(
    "Azure model '{model_name}' requires the 'endpoint' parameter. See https://docs.spiceai.org/components/models/azure for details."
).into(),
```

File: crates/runtime/src/model/chat.rs [hunk 3]
```diff
@@ -200,10 +200,19 @@ fn azure(
...
+        });
+    }
+
     if api_key.is_some() && entra_token.is_some() {
         return Err(LlmError::FailedToLoadModel {
             source: format!(
-                "For azure model '{model_name}', only one of 'azure_api_key' or 'azure_entra_token' can be provided."
+                "For azure model '{model_name}', only one of 'azure_api_key' or 'azure_entra_token' can be provided.\nFor details on model configuration, visit https://docs.spiceai.org/components/models/azure"
```
Comment (lukekim) [inline crates/runtime/src/model/chat.rs]:
```suggestion
                "Azure model '{model_name}' allows only one of 'azure_api_key' or 'azure_entra_token'. See https://docs.spiceai.org/components/models/azure for details."
```

File: crates/runtime/src/model/chat.rs [hunk 4]
```diff
@@ -212,7 +221,7 @@ fn azure(
     if api_key.is_none() && entra_token.is_none() {
         return Err(LlmError::FailedToLoadModel {
             source: format!(
-                "Azure model '{model_name}' requires 'azure_api_key' or 'azure_entra_token'."
+                "Azure model '{model_name}' requires 'azure_api_key' or 'azure_entra_token'.\nFor details on model configuration, visit https://docs.spiceai.org/components/models/azure"
```
Comment (lukekim) [inline crates/runtime/src/model/chat.rs]:
```suggestion
                "Azure model '{model_name}' requires either 'azure_api_key' or 'azure_entra_token'. See https://docs.spiceai.org/components/models/azure for details."
```

**Review by lukekim — DISMISSED** [review]

File: crates/runtime/src/model/chat.rs [hunk 5]
```diff
@@ -191,7 +191,9 @@ fn azure(
 ) -> Result<Box<dyn Chat>, LlmError> {
     let Some(model_name) = model_id else {
         return Err(LlmError::FailedToLoadModel {
-            source: format!("For Azure model '{model_name}', model id must be specified in `from:azure:<model_id>`.").into(),
+            source: format!(
+    "Azure model '{model_name}' requires a model ID in the format `from:azure:<model_id>`. See https://docs.spiceai.org/components/models/azure for details."
```
Comment (peasee) [inline crates/runtime/src/model/chat.rs:195]:
```suggestion
    "Azure model '{model_name}' requires a model ID in the format `from:azure:<model_id>`.\nFor details, visit: https://docs.spiceai.org/components/models/azure"
```

Links with additional context [should be on newlines](https://github.com/spiceai/spiceai/blob/trunk/docs/dev/error_handling.md). Let's also align with the existing wording where links appear like `For details, visit: ...`

File: crates/runtime/src/model/chat.rs [hunk 6]
```diff
@@ -200,10 +202,18 @@ fn azure(
     let api_key = extract_secret!(params, "azure_api_key");
     let entra_token = extract_secret!(params, "azure_entra_token");
 
+    if api_base.is_none() {
+        return Err(LlmError::FailedToLoadModel {
+            source: format!(
+    "Azure model '{model_name}' requires the 'endpoint' parameter. See https://docs.spiceai.org/components/models/azure for details."
```
Comment (peasee) [inline crates/runtime/src/model/chat.rs:208]:
```suggestion
    "Azure model '{model_name}' requires the 'endpoint' parameter.\nFor details, visit: https://docs.spiceai.org/components/models/azure"
```

File: crates/runtime/src/model/chat.rs [hunk 7]
```diff
@@ -200,10 +202,18 @@ fn azure(
...
+).into(),
+        });
+    }
+
     if api_key.is_some() && entra_token.is_some() {
         return Err(LlmError::FailedToLoadModel {
             source: format!(
-                "For azure model '{model_name}', only one of 'azure_api_key' or 'azure_entra_token' can be provided."
+                "Azure model '{model_name}' allows only one of 'azure_api_key' or 'azure_entra_token'. See https://docs.spiceai.org/components/models/azure for details."
```
Comment (peasee) [inline crates/runtime/src/model/chat.rs:216]:
```suggestion
                "Azure model '{model_name}' allows only one of 'azure_api_key' or 'azure_entra_token'.\nFor details, visit: https://docs.spiceai.org/components/models/azure"
```

File: crates/runtime/src/model/chat.rs [hunk 8]
```diff
@@ -212,7 +222,7 @@ fn azure(
     if api_key.is_none() && entra_token.is_none() {
         return Err(LlmError::FailedToLoadModel {
             source: format!(
-                "Azure model '{model_name}' requires 'azure_api_key' or 'azure_entra_token'."
+                "Azure model '{model_name}' requires either 'azure_api_key' or 'azure_entra_token'. See https://docs.spiceai.org/components/models/azure for details."
```
Comment (peasee) [inline crates/runtime/src/model/chat.rs:225]:
```suggestion
                "Azure model '{model_name}' requires either 'azure_api_key' or 'azure_entra_token'.\nFor details, visit: https://docs.spiceai.org/components/models/azure"
```

**Review by lukekim — APPROVED** [review]

**Pull request merged**